cmd/fleet/serve.go

```diff
@@ -498,7 +499,11 @@ the way that the Fleet server works.
 
 			var mdmPushService push.Pusher
 			nanoMDMLogger := service.NewNanoMDMLogger(kitlog.With(logger, "component", "apple-mdm-push"))
-			pushProviderFactory := buford.NewPushProviderFactory()
+			pushProviderFactory := buford.NewPushProviderFactory(buford.WithNewClient(func(cert *tls.Certificate) (*http.Client, error) {
+				return fleethttp.NewClient(fleethttp.WithTLSClientConfig(&tls.Config{
+					Certificates: []tls.Certificate{*cert},
```

Comment:
This is the APNs push certificate, this is exactly the same as before, just that now we create our own http client instead of using the default one of the `buford` package: https://github.com/fleetdm/fleet/blob/b6c51139bb20951456cd73560211941677ff4913/server/mdm/nanomdm/push/service/service.go#L65-L76

Reply:
In NewPushProvider:

https://github.com/fleetdm/fleet/blob/b6c51139bb20951456cd73560211941677ff4913/server/mdm/nanomdm/push/buford/buford.go#L64-L81

So, before it called `client, err = bufordpush.NewClient(*cert)` and created a non-proxy-supporting http client, now it calls our provided `newClientCallback` via `client, err = f.newClientCallback(cert)`, but the `cert` is exactly the same as before.

Reply:
Makes sense. Thanks!